odl/trafos/wavelet.py

```diff
@@ -21,9 +21,836 @@
 from __future__ import print_function, division, absolute_import
 from future import standard_library
 standard_library.install_aliases()
+from builtins import range, str, super
 
 # External
+import numpy as np
 
 # Internal
+from odl.discr.lp_discr import DiscreteLp
+from odl.operator.operator import Operator
 
-__all__ = ()
+try:
+    import pywt
+    WAVELET_AVAILABLE = True
+except ImportError:
+    WAVELET_AVAILABLE = False
+
+__all__ = ('DiscreteWaveletTrafo', 'DiscreteWaveletTrafoAdjoint',
+           'DiscreteWaveletTrafoInverse', 'WAVELET_AVAILABLE')
+
+
+_SUPPORTED_IMPL = ('pywt',)
+
+
+def list_of_coeff_sizes(shape, nscale, wbasis, mode):
+    """Construct a size list from given wavelet coefficients.
+
+    Construct a list containing the sizes of the wavelet approximation
+    and detail coefficients when wavelet basis, number of scaling levels
+    and the shape of the original image are given.
+    Related to 2D and 3D multidimensional wavelet transform.
+
+    Parameters
+    ----------
+    shape : `tuple`
+        Number of pixels/voxels in the image. Its lenght must be 2 or 3.
+    nscale : `int`
+        Number of scales in the multidimensional wavelet
+        transform
```

Comment:
The code checks if this value is smaller than the maximum number of scales determined by some `pywt` helper. If it's not possible to quickly state the constraint, you should say that this parameter is checked against the maximum number of scales returned by `pywt.dwt_max_level` and add a link to the online doc of this function if possible.